pkgs/top-level/all-packages.nix

```diff
@@ -6019,7 +6019,9 @@ with pkgs;
 
   fcitx5-table-other = callPackage ../tools/inputmethods/fcitx5/fcitx5-table-other.nix { };
 
-  fcppt = callPackage ../development/libraries/fcppt { };
+  fcppt = callPackage ../development/libraries/fcppt {
+    stdenv = if stdenv.targetPlatform.isAarch64 then gcc10Stdenv else stdenv;
```

Comment:
I'm not very familiar with c++, but changing the stdenv in a c++ library will make it incompatible with other c++ libraries, right ? So it would be impossible to use fcppt with boost (which is an intented use case as far as I can tell)

I suspect you chose gcc10 because aarch64 is still on gcc 9 which does not support all of c++20 yet, in this case there does not seem to be any good solution (apart from keeping the current version for aarch64).

In the end you are the maintainer so I leave the decision up to you.

Reply:
You are absolutely correct, I just wanted to see if I could actually get it to compile this way. But it's not a good solution indeed, so I reverted the change.